lib/facter/framework/formatters/yaml_fact_formatter.rb

```diff
@@ -15,7 +15,7 @@ def format(resolved_facts)
                            FormatterHelper.retrieve_facts_to_display_for_user_query(user_queries, resolved_facts)
                          end
 
-      facts_to_display = Psych.parse_stream(facts_to_display.to_yaml)
+      facts_to_display = Psych.parse_stream(facts_to_display.deep_to_h.to_yaml)
```

Comment:
does it helps to keep the inherited `to_yaml` or would be better to rewrite it to call `deep_to_h.to_yaml` and make `deep_to_h` private?